.github/actions/save-logs/action.yaml

```
@@ -44,37 +44,25 @@ runs:
       shell: bash
 
     - name: Move Playwright Artifacts
-      if: ${{ inputs.distro == 'k3d' }} # Currently only run on k3d
       run: |
         sudo mkdir -p /tmp/playwright
         sudo mv test/playwright/.playwright/* /tmp/playwright || true
       shell: bash
 
-    # Additional/specific debug for non-k3d clusters
-    - name: Pepr Debug
+    # Loop over all pods and use kubectl logs since we don't have the same direct node access as k3d
+    - name: Cluster Pod Logs
       if: ${{ inputs.distro != 'k3d' }}
       run: |
-        echo "::group::Pepr Pod Status and Metrics"
-        uds zarf tools kubectl top pods -n pepr-system
-        uds zarf tools kubectl get pods -n pepr-system
-        echo "::endgroup::"
-        echo "::group::Fetch pepr logs"
-        uds zarf tools kubectl logs -n pepr-system -l app=pepr-uds-core --tail -1 > /tmp/pepr-logs.log
-        uds zarf tools kubectl logs -n pepr-system -l app=pepr-uds-core --tail -1 --previous > /tmp/pepr-previous-logs.log || true
-        uds zarf tools kubectl logs -n pepr-system -l app=pepr-uds-core-watcher --tail -1 > /tmp/pepr-watcher-logs.log
-        uds zarf tools kubectl logs -n pepr-system -l app=pepr-uds-core-watcher --tail -1 --previous > /tmp/pepr-watcher-previous-logs.log || true
-        echo "::endgroup::"
-        echo "::group::Describe Failed Packages"
-        FAILED_PACKAGES=($(uds zarf tools kubectl get package -A -o jsonpath="{range .items[?(@.status.phase!='Ready')]}{.metadata.name}{','}{.metadata.namespace}{'\n'}{end}")); for PACKAGE in "${FAILED_PACKAGES[@]}"; do PACKAGE_NAME=$(echo "$PACKAGE" | awk -F "," '{print $1}'); PACKAGE_NAMESPACE=$(echo "$PACKAGE" | awk -F "," '{print $2}'); uds zarf tools kubectl describe package "$PACKAGE_NAME" -n "$PACKAGE_NAMESPACE"; echo; done
-        echo "::endgroup::"
-      shell: bash
-
-    - name: Keycloak Debug
-      if: ${{ inputs.distro != 'k3d' }}
-      run: |
-        echo "::group::Fetch Keycloak logs"
-        uds zarf tools kubectl logs sts/keycloak -n keycloak --tail -1 > /tmp/keycloak-logs.log
-        uds zarf tools kubectl logs sts/keycloak -n keycloak --tail -1 --previous > /tmp/keycloak-previous-logs.log || true
+        echo "::group::Collecting logs for all pods"
+        mkdir -p /tmp/pod-logs
+        # Loop over all pods and grab current and previous logs (8x parallel processes for efficiency)
+        uds zarf tools kubectl get pods -A -o jsonpath='{range .items[*]}{.metadata.namespace}{" "}{.metadata.name}{"\n"}{end}' \
+          | xargs -P 8 -n 2 bash -c '
+            ns="$1"; pod="$2"
+            base="/tmp/pod-logs/${ns}-${pod}"
+            uds zarf tools kubectl logs -n "$ns" "$pod" --tail=-1 > "${base}.log" 2>&1 || true
+            uds zarf tools kubectl logs -n "$ns" "$pod" --previous --tail=-1 > "${base}-previous.log" 2>&1 || true
```

grabbing logs for all containers might be helpful.  troubleshooting the loki sidecar configmap issue and that is a separate container that i wouldn't get with just this command
```suggestion
            uds zarf tools kubectl logs --all-containers=true -n "$ns" "$pod" --tail=-1 > "${base}.log" 2>&1 || true
            uds zarf tools kubectl logs --all-containers=true -n "$ns" "$pod" --previous --tail=-1 > "${base}-previous.log" 2>&1 || true
```

---

or you could loop over the containers in each pod and create separate files if we don't want to merge all container logs into a single file